resolver/from_json.py

```diff
@@ -0,0 +1,35 @@
+import json
+
+from typing import Union, Any
+
+from sceptre.exceptions import SceptreException
+from sceptre.resolvers import Resolver
+
+
+class FromJsonResolver(Resolver):
+    def resolve(self):
+        """
+        resolve is the method called by Sceptre. It should carry out the work
+        intended by this resolver. It should return a string to become the
+        final value.
+        """
+        return from_json(self.argument)
+
+
+def from_json(arg: Union[str, list]) -> Any:
+    try:
+        if isinstance(arg, str):
+            return json.loads(arg)
+
+        if isinstance(arg, list):
+            arg_count = len(arg)
+            if arg_count != 1:
+                raise SceptreException(
+                    f"!from_json expects exactly one argument, got {arg_count}"
+                )
+            return from_json(arg[0])
```

Comment:
@jfalkenstein What's the correct behavior if the first arg is also a list? Here I'm naïvely recursing in but I suspect the right thing is to confirm it's a string argument before recursing. (Recursion might not actually be necessary in that case)

Reply:
Yeah, no need to recurse here. Since you're creating this, you have two options on what to do and you could choose which you'd prefer.

Option 1 (the simplest): [Do like CloudFormation did with `!Not`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-conditions.html#intrinsic-function-reference-conditions-not) and require that the argument to these two resolvers is always and exclusively a single-element list. This has the aesthetics of almost looking like a function invocation with the brackets (i.e. `!from_json [ !request https://www.google.com/get-me-some-json ]`. This is simple because there isn't any question of how to pass the argument... it's always passed as a single-element list. I think 95% of uses of this resolver is going to be to wrap the output of other resolvers anyway, in which case you'll NEED to use a list (because Yaml doesn't support doing `!from_json !request ...`).

Option 2 (the most flexible): If the argument is a string, interpret that as the json to load; If the argument is a list, take the first item in that list as the argument to load. I think this approach, while more flexible, is probably silly, since that will just lead to confusion and almost everyone using this will want to use it with other resolvers anyway. This also has the added problem of how you do this in the reverse in `to_json`... because you can jsonify a list just like anything else.... there would be no way to know whether the user is intending to jsonify a dict that was passed in a list argument or to jsonify a list with a dict in it. 

For these reasons, I believe option 1 is definitely the right way to go because it's simpler, easier to document, and can be applied both in `to_json` and `from_json` in the same way, making everything simpler.